lib/elixir/lib/module/types/unify.ex

```diff
@@ -28,47 +30,29 @@ defmodule Module.Types.Unify do
   #   * subtype? (subtypes only)
   #   * has_unbound_var? (composite only)
   #   * recursive_type? (composite only)
-  #   * collect_vars (composite only)
+  #   * collect_var_indexes (composite only)
   #   * lift_types (composite only)
+  #   * expand_union (composite only)
+  #   * walk (composite only)
```

Comment:
I believe has_unbound_var? and recursive_type? can be implemented on top of `walk` now. We just need to `throw` if we find something. `collect_var_indexes` may be doable the same way.

Reply:
recursive_type? is not simpler to implement with `walk` because it needs to know about all composite types.

Reply:
I see. Your call on all of those. :)